File: 01-Assignment-02-02-2026/Using_C/program_17.c
/* Program 17: Find the roots of a quadratic equation */

#include <stdio.h>
#include <math.h>

int main() {
    double a, b, c, x1, x2, d;

    printf("For the equation: ax^2 + bx + c = 0\n");

    printf("Enter value of a : ");
    scanf("%lf", &a);

    printf("Enter value of b : ");
    scanf("%lf", &b);

    printf("Enter value of c : ");
    scanf("%lf", &c);

    if (a == 0) {
        printf("Not a quadratic equation");
        return 0;
    }

    d = b * b - 4 * a * c;

    if (d < 0) {
        printf("Roots are imaginary");
    }
    else {
        x1 = (-b + sqrt(d)) / (2 * a);
        x2 = (-b - sqrt(d)) / (2 * a);
        printf("Roots are:\nx1 = %.2lf\nx2 = %.2lf", x1, x2);
    }

    return 0;
}

File: 01-Assignment-02-02-2026/Using_Java/Program17.java
/*
Program 17
Java program to find the roots of a quadratic equation
*/

import java.util.Scanner;

public class Program17 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Quadratic Equation : aX^2 + bX + c = 0");
        System.out.print("a ? => ");
        double a = scanner.nextDouble();
        System.out.print("b ? => ");
        double b = scanner.nextDouble();
        System.out.print("c ? => ");
        double c = scanner.nextDouble();
        scanner.close();
        double d = Math.pow(b, 2) - (4 * a * c);
        if (d > 0) {
            System.out.println("2 real solutons exist");
            double x1 = (-b + (Math.sqrt(d))) / (2 * a);
            double x2 = (-b - (Math.sqrt(d))) / (2 * a);
            System.out.println("Roots -> " + x1 + ' ' + x2);
        } else if (d == 0) {
            double x1 = (-b + (Math.sqrt(d))) / (2 * a);
            double x2 = (-b - (Math.sqrt(d))) / (2 * a);
            System.out.println("1 real solution exist");
            System.out.println("Roots -> " + x1 + ' ' + x2);
        } else {
            System.out.println("0 real soultion exist");
        }

    }

}
